packages/phone-number-privacy/monitor/src/test.ts

```diff
@@ -86,38 +86,66 @@ export async function concurrentRPSLoadTest(
     | CombinerEndpointPNP.PNP_QUOTA
     | CombinerEndpointPNP.PNP_SIGN = CombinerEndpointPNP.PNP_SIGN,
   duration: number = 0,
-  bypassQuota?: boolean,
-  useDEK?: boolean
+  bypassQuota: boolean = false,
+  useDEK: boolean = false,
+  movingAverageRequests: number = 50
 ) {
-  const taskFn = async (i: number) => {
-    const start = performance.now()
+  const latencyQueue: number[] = []
+  let movingAvgLatencySum = 0
+  let latencySum = 0
+  let index = 1
+
+  function measureLatency(fn: () => Promise<void>): () => Promise<void> {
+    return async () => {
+      const start = performance.now()
+
+      await fn()
+
+      const reqLatency = performance.now() - start
+      latencySum += reqLatency
+      movingAvgLatencySum += reqLatency
+
+      const queuelength = latencyQueue.push(reqLatency)
+      if (queuelength > movingAverageRequests) {
```

Comment:
i would only compute the avg if and only if we have at least 50 or whatever number of elements we define... I'm not sure if says much it not.

You can ask for a few moving avgs... so [10,50,100]